lib/bloc/podcast/podcast_bloc.dart

```diff
@@ -436,13 +444,26 @@ class PodcastBloc extends Bloc {
   }
 
   void applySearchFilter() {
-    if (_searchTerm.isEmpty) {
-      _episodesStream.add(_episodes);
-    } else {
-      var searchFilteredEpisodes =
-          _episodes.where((e) => e.title!.toLowerCase().contains(_searchTerm.trim().toLowerCase())).toList();
-      _episodesStream.add(searchFilteredEpisodes);
+    List<Episode> filtered = _episodes;
```

Comment:
Filtering the episodes here is the wrong place to do it. It needs to be handled at the database level.